File: 2248-intersection-of-multiple-arrays/2248-intersection-of-multiple-arrays.java
class Solution {
    public List<Integer> intersection(int[][] arr) {
        HashMap<Integer,Integer> map=new HashMap<>();
        for(int i=0;i<arr.length;i++){
            for(int j=0;j<arr[i].length;j++){
                if(!map.containsKey(arr[i][j])){
                    map.put(arr[i][j],1);
                }
                else{
                    map.put(arr[i][j],map.get(arr[i][j])+1);
                }
            }
        }
        List<Integer> list=new ArrayList<>();
        for(int i:map.keySet()){
            if(map.get(i)==arr.length){
                list.add(i);
            }
        }
        Collections.sort(list);
        return list;
    }
}

File: 2248-intersection-of-multiple-arrays/2248-intersection-of-multiple-arrays.py
import itertools
from collections import Counter
class Solution:
    def intersection(self, nums: List[List[int]]) -> List[int]:
        b=list(itertools.chain.from_iterable(nums))
        c=Counter(b)
        max_value=max(c.values())
        l=[]
        if max_value==len(nums):
            for i in c.keys():
                if c.get(i)==max_value:
                    l.append(i)
            l.sort()
        return l